File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic/SourceModuleBinder.cs
using System.Runtime.InteropServices;
using Microsoft.CodeAnalysis.VisualBasic.Symbols;

namespace Microsoft.CodeAnalysis.VisualBasic
{
	internal class SourceModuleBinder : Binder
	{
		private readonly SourceModuleSymbol _sourceModule;

		public override OptionStrict OptionStrict => _sourceModule.Options.OptionStrict;

		public override bool OptionInfer => _sourceModule.Options.OptionInfer;

		public override bool OptionExplicit => _sourceModule.Options.OptionExplicit;

		public override bool OptionCompareText => _sourceModule.Options.OptionCompareText;

		public override bool CheckOverflow => _sourceModule.Options.CheckOverflow;

		public override QuickAttributeChecker QuickAttributeChecker => _sourceModule.QuickAttributeChecker;

		public SourceModuleBinder(Binder containingBinder, SourceModuleSymbol sourceModule)
			: base(containingBinder, sourceModule, sourceModule.ContainingSourceAssembly.DeclaringCompilation)
		{
			_sourceModule = sourceModule;
		}

		public override AccessCheckResult CheckAccessibility(Symbol sym, [In][Out] ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, TypeSymbol accessThroughType = null, BasesBeingResolved basesBeingResolved = default(BasesBeingResolved))
		{
			if (!base.IgnoresAccessibility)
			{
				return AccessCheck.CheckSymbolAccessibility(sym, _sourceModule.ContainingSourceAssembly, ref useSiteInfo, basesBeingResolved);
			}
			return AccessCheckResult.Accessible;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/SourceModuleBinder.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic
	Friend Class SourceModuleBinder
		Inherits Binder
		Private ReadOnly _sourceModule As SourceModuleSymbol

		Public Overrides ReadOnly Property CheckOverflow As Boolean
			Get
				Return Me._sourceModule.Options.CheckOverflow
			End Get
		End Property

		Public Overrides ReadOnly Property OptionCompareText As Boolean
			Get
				Return Me._sourceModule.Options.OptionCompareText
			End Get
		End Property

		Public Overrides ReadOnly Property OptionExplicit As Boolean
			Get
				Return Me._sourceModule.Options.OptionExplicit
			End Get
		End Property

		Public Overrides ReadOnly Property OptionInfer As Boolean
			Get
				Return Me._sourceModule.Options.OptionInfer
			End Get
		End Property

		Public Overrides ReadOnly Property OptionStrict As Microsoft.CodeAnalysis.VisualBasic.OptionStrict
			Get
				Return Me._sourceModule.Options.OptionStrict
			End Get
		End Property

		Public Overrides ReadOnly Property QuickAttributeChecker As Microsoft.CodeAnalysis.VisualBasic.Symbols.QuickAttributeChecker
			Get
				Return Me._sourceModule.QuickAttributeChecker
			End Get
		End Property

		Public Sub New(ByVal containingBinder As Binder, ByVal sourceModule As SourceModuleSymbol)
			MyBase.New(containingBinder, sourceModule, sourceModule.ContainingSourceAssembly.DeclaringCompilation)
			Me._sourceModule = sourceModule
		End Sub

		Public Overrides Function CheckAccessibility(ByVal sym As Symbol, <InAttribute> <Out> ByRef useSiteInfo As CompoundUseSiteInfo(Of AssemblySymbol), Optional ByVal accessThroughType As TypeSymbol = Nothing, Optional ByVal basesBeingResolved As Microsoft.CodeAnalysis.VisualBasic.BasesBeingResolved = Nothing) As AccessCheckResult
			If (MyBase.IgnoresAccessibility) Then
				Return AccessCheckResult.Accessible
			End If
			Return AccessCheck.CheckSymbolAccessibility(sym, Me._sourceModule.ContainingSourceAssembly, useSiteInfo, basesBeingResolved)
		End Function
	End Class
End Namespace